src/components/TodoList.tsx

```diff
@@ -0,0 +1,62 @@
+import { CSSTransition, TransitionGroup } from 'react-transition-group';
+import { useContext } from 'react';
+import { TodoItem } from './TodoItem';
+import { TodosContext } from './TodosContext';
+
+export const TodoList: React.FC = () => {
+  const { visibleTodos, tempTodo } = useContext(TodosContext);
+
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      <TransitionGroup>
+        {visibleTodos.map(todo => (
+          <CSSTransition
+            key={todo.id}
+            timeout={300}
+            classNames="item"
+          >
+            <TodoItem
+              key={todo.id}
+              todo={todo}
+            />
+
+          </CSSTransition>
+        ))}
+        {tempTodo && (
+          <CSSTransition
+            key={tempTodo.id}
+            timeout={300}
+            classNames="temp-item"
+          >
+            <div data-cy="Todo" className="todo">
+              <label className="todo__status-label">
+                <input
+                  data-cy="TodoStatus"
+                  type="checkbox"
+                  className="todo__status"
+                />
+              </label>
+              <span data-cy="TodoTitle" className="todo__title">
+                {tempTodo.title}
+              </span>
+              <button
+                type="button"
+                className="todo__remove"
+                data-cy="TodoDelete"
+              >
+                ×
+              </button>
+              <div
+                data-cy="TodoLoader"
+                className="modal overlay is-active"
+              >
+                <div className="modal-background has-background-white-ter" />
+                <div className="loader" />
+              </div>
+            </div>
```

Comment:
looks like you can write the TodoItem component here and just pass some prop that will show the loader